zip-0227.rst

```diff
@@ -1,8 +1,359 @@
+
+2. The `issuanceValidatingKey`, denoted as `ik`, is the key that is used to validate the issuance transaction. This key is used to validate the issuance of a specific asset type, and is used by all blockchain users (specifically the asset owners and consensus validators) to associate the asset in question with the issuer. Throughout we also call this key the issuer verification key.
+
+The two keys are derived in an analogous manner to the `spendAuthorizingKey` and `spendValidatingKey` keys, using the same signature scheme, as described in ZIP 32 [#zip-0032]_ and in accordance with ZIP 316 [#zip-0316]_.
+
+- The `issuanceAuthorizingKey` is derived directly from the `spendingKey`, `sk`, as a private signature key:
+
+:math:`\mathsf{isk := ToScalar^{Orchard}(︀ PRF^{expand}_{sk} ([0x0a])}`
+
+- The `issuanceValidatingKey` is derived from the `issuanceAuthorizingKey`, `isk`, as a public verification key:
+
+:math:`\mathsf{ik := SpendAuthSig^{Orchard}.DerivePublic(isk)}`
+
+This allows the issuer to use the same wallet it usually uses to transfer assets, while keeping a disconnect from the other keys. Specifically, this method is aligned with the requirements and motivation of ZIP 32 [#zip-0032]_, and can further be in  It provides further anonymity and the ability to delegate issuance of an asset (or in the future, generate a multi-signature protocol) while the rest of the keys remain in the wallet safe.
+
+Asset Identifier
+----------------
+
+For every new asset, there must be a new and unique identifier of the asset. Every asset is defined by the asset description, `asset_desc`, which is a global byte string (valid across all -future- Zcash protocols). From this asset descritpion, the specific asset identifier, `AssetId`, is derived with a `GroupHash` and used within the output notes, as defined by the specific protocol (e.g.: for now Orchard-style, using Pallas curve).
+
+The main requirement for the asset identifier is for it to be collision-free, as two different issuers MUST not be able to issue the same asset type. One way to achieve this is to make the identifier dependent on the issuer keys, as the issuer keys are unique to each issuer. We achieve this by deriving the asset identifier, `AssetId` as follows:
+
+:math:`\mathsf{AssetId := GroupHash^{\mathbb{P}}}\mathsf{(ik || asset_desc)}`
+
+where
+
+- `asset_desc` is the asset description, a UTF-8 encoded string of up to 512 bytes, which includes any information pertaining to the issuance 
+- `issuanceValidatingKey` is the public key of the issuer, used to verify the signature on the transaction SIGHASH and is defined as `ik`
+
+Global Issuance State
+---------------------
+
+Issuance requires the following additions to the global state: 
+- `previously_finalized`, a set of `AssetId` that have been finalized (i.e.: the `finalize` flag has been set to `1`).
```

Comment:
Do we also need to store a global list of `AssetId`s or `asset_desc` that have been previously issued?

How will the same assets be recognised across pools, if we don't store `asset_desc`?

Reply:
Storing `assetId`s and `asset_desc` for non finalized assets is not required for properly verify consensus.
However, this information is very relevant and can be tracked by the client as part of an analytics service. 

Across polls, assets are uniquely identified by the `assetId`(which is driven from `asset_desc`), so I don't think we need anything else.

Reply:
Does this commit us to always using the Pallas group hash, even if we deprecate Orchard?

Reply:
Specifically, the Orchard issuance action includes the `asset_desc`, not the `AssetId`, so we will always need the Pallas hash to be able to convert `asset_desc` to `AssetId` for verification.

Reply:
The design has now been updated such that the `AssetId` is global, even across possible future shielded pools, with hashing done into the Orchard shielded pool at present. In future, a different hash could be used for a different pool.